src/TestFramework/Coverage/XmlReport/IndexXmlCoverageParser.php

```diff
@@ -97,6 +97,12 @@ private function parseNodes(
      */
     private static function assertHasExecutedLines(SafeDOMXPath $xPath, bool $isForGitDiffLines): void
     {
+        $testsNode = $xPath->query('/phpunit/project/tests')->item(0);
+
+        if ($testsNode !== null && !$testsNode->hasChildNodes()) {
+            throw NoTestsInProject::create();
```

Comment:
Unfortunately, this is not an ideal solution, because it conflicts with the code below, when no tests executed by some mistake in `@covers()` or `--group` or `--filter`.

The only one correct way of determining if PHPUnit "sees" test files in the project is to use `phpunit --list-tests`, but I don't like to run one more process each time to cover this rare case, so I think it doesn't worth it.

Bootstrap project can have an example test and then run Infection, and that's it.